org.eclipse.transformer/src/main/java/org/eclipse/transformer/action/impl/ChangesImpl.java

```diff
@@ -101,15 +105,30 @@ public void log(Logger logger, String inputPath, String outputPath) {
 				logger.info(consoleMarker, "Input  [ {} ] as [ {} ]", getInputResourceName(), inputPath);
 			}
 			if (useOutputName.equals(outputPath)) {
-				logger.info(consoleMarker, "Output [ {} ]", outputPath);
+				logger.info(consoleMarker, "Output [ {} ] took [ {} ] ms", outputPath,
+							toHoursMinutesSeconds(getElapsedMillis()));
 			} else {
-				logger.info(consoleMarker, "Output [ {} ] as [ {} ]", getOutputResourceName(), outputPath);
+				logger.info(consoleMarker, "Output [ {} ] as [ {} ] took [ {} ] ms", getOutputResourceName(), outputPath,
+							toHoursMinutesSeconds(getElapsedMillis()));
 			}
 
 			logChanges(logger);
 		}
 	}
 
+	public static String toHoursMinutesSeconds(final long ms) {
```

Comment:
Shouldn't this use DateTimeFormatter instead of custom code?

`DateTimeFormatter.ISO_LOCAL_TIME.format(Duration.ofNanos(System.nanoTime() - start));`

Reply:
No you can't do that.
The best we can do is upgrade to Java 9 at least (I picked 11 and use duration)

> final Duration duration = Duration.ofMillis(millis);
> final long HH = duration.toHours();
> final long MM = duration.toMinutesPart();
> final long SS = duration.toSecondsPart();
> return String.format("%02d:%02d:%02d", HH, MM, SS);

It works to replace a small and easy computation. It's a taste question here. I don't mind 3 lines of a simple computation instead of using a complex object sometimes.

Mind that the first one is really toHours() and not toHoursPart()

We could also do it using something like

> long HH = TimeUnit.MILLISECONDS.toHours(millis);
> long MM = TimeUnit.MILLISECONDS.toMinutes(millis) % 60;
> long SS = TimeUnit.MILLISECONDS.toSeconds(millis) % 60;

I pushed the java upgrade and the duration version